agent/node.go

```diff
@@ -633,7 +634,23 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			ready = nil
 		}
 
-		err = n.waitRole(ctx, ca.AgentRole)
+		roleChanged := make(chan struct{})
+		go func() {
+			n.waitRole(ctx, ca.AgentRole)
+			roleChanged <- struct{}{}
+		}()
+
+		select {
+		case <-done:
+			// Fail out if m.Run() returns error, otherwise wait for
+			// role change.
+			if runErr != nil {
+				err = runErr
+			} else {
+				<-roleChanged
```

Comment:
@aaronlehmann @LK4D4 I'm not familiar with manager enough to know if this is safe. Is demotion only reason manager can return without an error?